### ISSUE TITLE
compilation of SCSS theme takes too much time

### ISSUE THREAD

**Issue by svetq** [telerik/kendo-themes#423]
## I'm submitting a...

* Suggestion for improvement

## Current behavior

Building the SCSS theme takes around 30 seconds.

## Minimal reproduction of the problem with instructions
Create a start up Angular 7 project and load just the Kendo Default theme in angular.json as follows:

```
"styles": [
    "node_modules/@progress/kendo-theme-default/scss/all.scss"
]
```

run ng build --prod

## Environment
Product:
* Kendo Ui for Angular

**Comment by msmolka** [comment]
I wonder what takes so much time. Less compilation of those files is very quick. Native bootstrap one is very quick.

**Comment by joneff** [comment]
There are several factors that contribute to the slow build:

* we want things to be modular: for instance, if you want to import just the grid, then you import just the grid and things should work with just one line
* we use dependencies a lot. Like a lot of lot. Since there is no official way to do it, we use `@import` syntax.
* that leads us to the next problem: due to the inherent feature of sass, each time you include a file it may produce a different output. We've some one mitigated the issue using a-la `import_once` approach to reduce the output, but the files are still read.

At the end of the day, simply importing the grid will import dialog, button, common files ... All in all the compiler should process around a 100 files, with about 20 being unique and the rest a result of the recursive dependency loading.

Now extrapolate the same thing for the entire theme and things get beyond outrageously slow with the compiler easily processing well beyond thousands of files with a couple of hundred being unique.

Consolidating mixins and commonly used files will increase performance. Slightly increasing the complexity of imports, say a "core / min" file and then grid or window will improve things too. But the dependencies thing will remain.

That could be solved with additional tasks that are triggered by developers. For instance, we have our own package importer that plugs into sass build process and as a result, our build tasks is fairly quicker. However, we've not made to be consumed by the likes of webpack, per se.

To be completely honest, there is not silver bullet how to make things both modular and blazing fast.

**Comment by minimasci** [comment]
@joneff Hi, 

I am still experiencing some crazy compilation times (~100seconds).

I'm a little new to theming using vendor .scss files. I'm sure the "issue" is on my end - as I'm still trying to figure out how all of this works and how to increase performance. I'm currently using Visual Studio Code with the "Live Sass Compiler" extension. I have included "@progress/kendo-theme-bootstrap": "^4.36.0" in my package.json using node JS for importing source files. 

I have 3 levels of variables due to the complexity of our projects. But ultimately they're just a combination of custom variables and overwritten bootstrap/kendo variables for theming. Nothing too large and no mixins/functions/styles
![image](https://user-images.githubusercontent.com/14220537/129916647-9aa8c4d2-1d35-4f8a-a789-e5dbf5eac29f.png)

The _kendo.theme.scss is simply importing certain controls so as not to include the entire system (as I presume this way is faster?):
![image](https://user-images.githubusercontent.com/14220537/129916810-f4592b27-a901-43d1-a32e-36cf612ee2e8.png)


**Can you please assist by offering some additional instructions (to noobs like me) to help me understand how exactly to make use of your fix (#909) for this issue?** It's not clear, to me, how your fix made things fast and how to actually make use of it. Do we need to avoid third-party compilers? Do we need to import a specific .scss file?
